packages/babel-helper-create-class-features-plugin/src/fields.ts

```diff
@@ -188,9 +189,28 @@ const privateNameVisitor = privateNameVisitorFactory<
   },
 });
 
+// rename all bindings that shadows innerBinding
+function unshadow(
+  name: string,
+  scope: Scope,
+  innerBinding: t.Identifier | undefined,
+) {
+  // in some cases, scope.getBinding(name) === undefined
+  // so we check hasBinding to avoid keeping looping
+  // see: https://github.com/babel/babel/pull/13656#discussion_r686030715
+  while (
+    scope?.hasBinding(name) &&
+    !scope.bindingIdentifierEquals(name, innerBinding)
+  ) {
+    scope.rename(name);
```

Comment:
Can renaming ever fail? I can't think of a case myself.

Reply:
No, it always succeeds (but it might be observable since it sometimes changes the `.name` property of functions).